proxy/src/main/java/com/velocitypowered/proxy/command/VelocityCommandManager.java

```diff
@@ -56,8 +64,14 @@ public boolean execute(@NonNull CommandSource source, @NonNull String cmdLine) {
       if (!command.hasPermission(source, actualArgs)) {
         return false;
       }
-
-      command.execute(source, actualArgs);
+      CommandExecuteEvent event = new CommandExecuteEvent(source, alias, actualArgs);
+      server.getEventManager().fire(event)
+              .thenAcceptAsync(cee -> {
+                CommandExecuteEvent.CommandResult commandResult = cee.getResult();
+                if (commandResult.isAllowed()) {
+                  command.execute(source, actualArgs);
```

Comment:
This is incorrect, since it will not reflect the result a plugin may choose to set.

Reply:
What would the prefered result be ?

Reply:
Okay. Right now your `CommandResult` doesn't actually let people replace the command (although the fields and getters exist). 

Either add this functionality or remove the redundant fields.